<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8">
<title>Content-Centric Networking in Java: org.ccnx.ccn.test.ThreadAssertionTest Class Reference</title>
<link href="doxygen.css" rel="stylesheet" type="text/css">
<link href="tabs.css" rel="stylesheet" type="text/css">
</head><body>
<!-- Generated by Doxygen 1.5.6 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul>
      <li><a href="main.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Packages</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
    </ul>
  </div>
  <div class="tabs">
    <ul>
      <li><a href="annotated.html"><span>Class&nbsp;List</span></a></li>
      <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath"><b>org</b>.<b>ccnx</b>.<b>ccn</b>.<a class="el" href="namespaceorg_1_1ccnx_1_1ccn_1_1test.html">test</a>.<a class="el" href="classorg_1_1ccnx_1_1ccn_1_1test_1_1_thread_assertion_test.html">ThreadAssertionTest</a>
  </div>
</div>
<div class="contents">
<h1>org.ccnx.ccn.test.ThreadAssertionTest Class Reference</h1><!-- doxytag: class="org::ccnx::ccn::test::ThreadAssertionTest" -->Test cases for <a class="el" href="classorg_1_1ccnx_1_1ccn_1_1test_1_1_thread_assertion_runner.html" title="Junit cannot catch exceptions from spawned threads.">ThreadAssertionRunner</a>.  
<a href="#_details">More...</a>
<p>

<p>
<a href="classorg_1_1ccnx_1_1ccn_1_1test_1_1_thread_assertion_test-members.html">List of all members.</a><table border="0" cellpadding="0" cellspacing="0">
<tr><td></td></tr>
<tr><td colspan="2"><br><h2>Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top"><a class="anchor" name="dcf63a53ef6baef80c412ffb0317d3cf"></a><!-- doxytag: member="org::ccnx::ccn::test::ThreadAssertionTest::testNoError" ref="dcf63a53ef6baef80c412ffb0317d3cf" args="()" -->
void&nbsp;</td><td class="memItemRight" valign="bottom"><b>testNoError</b> ()  throws Exception </td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top"><a class="anchor" name="08cb24cb987524b1a2b49601b3e1e6de"></a><!-- doxytag: member="org::ccnx::ccn::test::ThreadAssertionTest::testAssertFail" ref="08cb24cb987524b1a2b49601b3e1e6de" args="()" -->
void&nbsp;</td><td class="memItemRight" valign="bottom"><b>testAssertFail</b> ()  throws Exception </td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top"><a class="anchor" name="15169d8f37e234a45dc8bd2b8583b928"></a><!-- doxytag: member="org::ccnx::ccn::test::ThreadAssertionTest::testAssertError" ref="15169d8f37e234a45dc8bd2b8583b928" args="()" -->
void&nbsp;</td><td class="memItemRight" valign="bottom"><b>testAssertError</b> ()  throws Exception </td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top"><a class="anchor" name="dd3dc99544495b7779da00f73d4fe272"></a><!-- doxytag: member="org::ccnx::ccn::test::ThreadAssertionTest::testNullPointer" ref="dd3dc99544495b7779da00f73d4fe272" args="()" -->
void&nbsp;</td><td class="memItemRight" valign="bottom"><b>testNullPointer</b> ()  throws Exception </td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top"><a class="anchor" name="4c3e149352e1757ee120280a393da6c7"></a><!-- doxytag: member="org::ccnx::ccn::test::ThreadAssertionTest::testRuntimeException" ref="4c3e149352e1757ee120280a393da6c7" args="()" -->
void&nbsp;</td><td class="memItemRight" valign="bottom"><b>testRuntimeException</b> ()  throws Exception </td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top"><a class="anchor" name="b8aab43b3fb447c10e04e37d802fd47e"></a><!-- doxytag: member="org::ccnx::ccn::test::ThreadAssertionTest::testNoErrorImmediate" ref="b8aab43b3fb447c10e04e37d802fd47e" args="()" -->
void&nbsp;</td><td class="memItemRight" valign="bottom"><b>testNoErrorImmediate</b> ()  throws Exception </td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top"><a class="anchor" name="26bce37b6c1e9b1cb34f421d50b4930c"></a><!-- doxytag: member="org::ccnx::ccn::test::ThreadAssertionTest::testAssertFailImmediate" ref="26bce37b6c1e9b1cb34f421d50b4930c" args="()" -->
void&nbsp;</td><td class="memItemRight" valign="bottom"><b>testAssertFailImmediate</b> ()  throws Exception </td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top"><a class="anchor" name="4a82a07b0790c53ccb3d3e5391735a6c"></a><!-- doxytag: member="org::ccnx::ccn::test::ThreadAssertionTest::testAssertErrorImmediate" ref="4a82a07b0790c53ccb3d3e5391735a6c" args="()" -->
void&nbsp;</td><td class="memItemRight" valign="bottom"><b>testAssertErrorImmediate</b> ()  throws Exception </td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top"><a class="anchor" name="8a047c42b4fc31c2d29d82b1f0c1a395"></a><!-- doxytag: member="org::ccnx::ccn::test::ThreadAssertionTest::testNullPointerImmediate" ref="8a047c42b4fc31c2d29d82b1f0c1a395" args="()" -->
void&nbsp;</td><td class="memItemRight" valign="bottom"><b>testNullPointerImmediate</b> ()  throws Exception </td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top"><a class="anchor" name="1348bdabe0a2dc1ca1b2a41160b990ff"></a><!-- doxytag: member="org::ccnx::ccn::test::ThreadAssertionTest::testRuntimeExceptionImmediate" ref="1348bdabe0a2dc1ca1b2a41160b990ff" args="()" -->
void&nbsp;</td><td class="memItemRight" valign="bottom"><b>testRuntimeExceptionImmediate</b> ()  throws Exception </td></tr>

<tr><td colspan="2"><br><h2>Classes</h2></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><b>InnerRunner</b></td></tr>

</table>
<hr><a name="_details"></a><h2>Detailed Description</h2>
Test cases for <a class="el" href="classorg_1_1ccnx_1_1ccn_1_1test_1_1_thread_assertion_runner.html" title="Junit cannot catch exceptions from spawned threads.">ThreadAssertionRunner</a>. 
<p>
Also shows how to use it.<p>
<a class="el" href="classorg_1_1ccnx_1_1ccn_1_1test_1_1_thread_assertion_runner.html" title="Junit cannot catch exceptions from spawned threads.">ThreadAssertionRunner</a> will capture any Exception or Error. We just use NullPointer and Runtime as two examples here. <hr>The documentation for this class was generated from the following file:<ul>
<li>src/org/ccnx/ccn/test/ThreadAssertionTest.java</ul>
</div>
<hr size="1"><address style="text-align: right;"><small>Generated on Wed Sep 14 12:44:32 2011 for Content-Centric Networking in Java by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img src="doxygen.png" alt="doxygen" align="middle" border="0"></a> 1.5.6 </small></address>
</body>
</html>
